NSSAgent/NSSAgent.cs

```diff
@@ -100,10 +100,15 @@ public interface INSSAgent:IAuthenticationAgent
 
         //Roles
         IQueryable<string> GetRoles();
-        
+
+        //Statuses
+        IQueryable<Status> GetStatuses();
+        Task<Status> GetStatus(Int32 ID);
+        IQueryable<Status> GetApplicableStatuses(Manager manager = null, bool isStreamStats = false);
+
         //Scenarios
         //IQueryable<Scenario> GetScenarios(List<string> regionList, List<string> regressionRegionList, List<string> statisticgroupList = null, List<string> regressionTypeIDList = null, List<string> extensionMethodList = null, Int32 systemtypeID = 0);
-        IQueryable<Scenario> GetScenarios(List<string> regionList=null, Geometry geom=null, List<string> regressionRegionList = null, List<string> statisticgroupList = null, List<string> regressionTypeIDList = null, List<string> extensionMethodList = null, Int32 systemtypeID = 0, Manager manager = null);
+        IQueryable<Scenario> GetScenarios(List<string> regionList=null, Geometry geom=null, List<string> regressionRegionList = null, List<string> statisticgroupList = null, List<string> regressionTypeIDList = null, List<string> extensionMethodList = null, Int32 systemtypeID = 0, Manager manager = null, bool isStreamStats = false);
```

Comment:
hmm, yeah I'm still not sold on the idea of having a streamstats 'flag' for configuration. I think a policy type of approach might be more sustainable long term.

Reply:
How would you recommend doing this? Right now, it's expecting a header "X-Is-Streamstats" sent through by the client.

Reply:
hmm, thats odd the rest of my comment didn't stay. I can't remember exactly what I wrote last time, but the thought would be to move most of the Streamstats  'configuration' stuff out of the agent and into the services. This would allow the agent to be more flexible and allow the services to provide the configuration. A lot of the configuration could be performed in a custom middleware pipeline method that could capture the client info and write the policy claim, similar to how we are doing roles, but this one could be specific to nssservices. Regarding  the [filtering](https://github.com/USGS-WiM/NSSServices/blob/KJ-114/NSSAgent/NSSAgent.cs#L561) instead of 'isStreamstats' supply a List of status, similar to how we are filtering the regions, regressionregions etc.... leaving the method more flexible, and not tied to a client variable....
Hopefully that makes sense